tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -478,3 +479,95 @@ def _check_rules(self, resource_rules):
             return self.policy_include.issubset(resource_rules)
         else:  # validated earlier, can never happen
             raise FilterValidationError("Internal error.")
+
+
+class ResourceLockFilter(Filter):
+    """
+    Filter locked resources.
+    Lock can be of 2 types: ReadOnly and CanNotDelete. To filter any lock, use "Any" type.
+    Lock type is optional, by default any lock will be applied to the filter.
+    To get unlocked resources, use "Absent" type.
+
+    :example: Get all keyvaults with ReadOnly lock:
+
+    .. code-block :: yaml
+
+       policies:
+        - name: locked-keyvaults
+          resource: azure.keyvault
+          filters:
+            - type: resource-lock
+              lock-type: ReadOnly
+
+    :example: Get all locked sqldatabases (any type of lock):
+
+    .. code-block :: yaml
+
+       policies:
+        - name: locked-sqldatabases
+          resource: azure.sqldatabase
+          filters:
+            - type: resource-lock
+
+    :example: Get all unlocked resource groups:
+
+    .. code-block :: yaml
+
+       policies:
+        - name: unlock-rgs
+          resource: azure.resourcegroup
+          filters:
+            - type: resource-lock
+              lock-type: Absent
+
+    """
+
+    schema = type_schema(
+        'resource-lock', required=['type'],
+        **{
+            'lock-type': {'enum': ['ReadOnly', 'CanNotDelete', 'Any', 'Absent']},
+        })
+
+    schema_alias = True
+
+    def __init__(self, data, manager=None):
+        super(ResourceLockFilter, self).__init__(data, manager)
+        self.lock_type = self.data.get('lock-type', 'Any')
+
+    def process(self, resources, event=None):
+        result, _ = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._process_resource_set,
+            executor_factory=self.executor_factory,
+            log=self.log
+        )
```

Comment:
We should handle exceptions if any (raise them in the main thread since this is a filter)